tools/blue_deployment/step1.js

```diff
@@ -0,0 +1,343 @@
+/**
+ * @typedef {Object} ChainInfo
+ * @property {string} chainId The chain id.
+ * @property {string} blockHeight The height (= block number) of the the
+ *                    most recent block of the chain.
+ * @property {string} stateRoot The state root of the the most recent
+ *                    block of the chain.
+ */
+
+/**
+ * @typedef {Object} Contract
+ */
+
+/**
+ * @typedef {Object} ContractReference
+ */
+
+/**
+ * @typedef {string} Address
+ */
+
+const Web3 = require('web3');
+const colors = require('colors/safe');
+
+const { Contract, ContractRegistry } = require('../deployment_tool');
+const { deployContracts, UnlockedWeb3Signer } = require('./helpers');
+
+// root directory of npm project
+const rootDir = `${__dirname}/../../`;
+
+/**
+ * Returns the provided address as-is or or deploy a new EIP20Token and return its
+ * address if the provided address is the special string "new".
+ *
+ * @param {string} rpcEndpoint RPC endpoint for the web3 instance to use.
+ * @param {Address} deployerAddress Address of the account to use for deployment.
+ * @param {Address|string} eip20Address Either an existing address of a EIP20Token,
+ *                 or the special string "new".
+ * @param {object} deployOptions See {@link deployContracts}.
+ *
+ * @returns {Address} The address of either the provided EIP20Token or the
+ *                   newly deployed one.
+ */
+const tryDeployNewToken = async (rpcEndpoint, deployerAddress, eip20Address, deployOptions) => {
```

Comment:
Why is this function called `try...`?

Reply:
`try...` is the closest naming pattern I know that signals "do some work only if the necessary conditions have not been met already". I'm open to suggestions here!

Reply:
Ah, now I see. It returns the existing one if it is deployed already. What about `deployedToken()`? 🤔